packages/x-data-grid/src/models/gridBaseSlots.ts

```diff
@@ -179,6 +179,26 @@ type ClickAwayMouseEventHandler =
   | 'onPointerUp';
 type ClickAwayTouchEventHandler = 'onTouchStart' | 'onTouchEnd';
 
+export type PaginationProps = {
+  count: number;
+  page: number;
+  rowsPerPage: number;
+  rowsPerPageOptions?: readonly (number | { value: number; label: string })[];
+  onPageChange: (event: React.MouseEvent<HTMLButtonElement> | null, page: number) => void;
+  onRowsPerPageChange?: (rowsPerPage: number) => void;
+
+  disabled?: boolean;
+
+  getItemAriaLabel?: (type: 'first' | 'last' | 'next' | 'previous') => string;
+  labelRowsPerPage?: string;
+  labelDisplayedRows?: (info: {
+    from: number;
+    to: number;
+    count: number;
+    page: number;
+  }) => React.ReactNode | undefined;
```

Comment:
Having this here is ugly, but I don't see how to avoid it.

Reply:
Tbd: move pagination locales from the core to here.

Reply:
> Having this here is ugly, but I don't see how to avoid it.

Alternatively, we can use `apiRef.current.getLocaleText()` instead of passing the prop.
But passing a prop looks good to me too.

Reply:
I've handled l10n in the material bindings, other design-systems can easily fork that code and it removes l10n props from the typings.